### DISCUSSION TITLE
sessionWebClient 사용법?

### DISCUSSION THREAD

**Discussion by punch5545** [storycraft/node-kakao#663] [Q&A] [answered]
안녕하세요

SessionWebClient로 node-kakao에서 구현되지 않은 api를 호출하고싶은데요.

createSessionWebClient 메서드에서 첫번째 인자인 credential과 두번째 인자인 config 뭘 넣어야하는지 감이 안잡히네요.

config에는 OAuthApiClient의 config을 받아서 넣으면 될 것 같긴 한데......

credential에는 뭘 넣어도 401을 리턴하네요. 혹시나해서 createSessionProxy로도 해봤지만.. 실패..

도움 부탁드립니다.

== 추가내용 == 

![image](https://user-images.githubusercontent.com/28100137/133918837-31057817-269a-46ab-a85e-2e21ab0450b1.png)

헤더는 잘 들어가는데도 401을 반환하네요.

**Comment by storycraft** [comment, marked as answer]
NodeKakao내에서 `createSessionWebClient` 메서드의 사용 예는 이곳에서 볼 수 있습니다. https://github.com/storycraft/node-kakao/blob/768cd3ae1659131ba8dfe8d3f5738afcf268bb62/src/api/service-api-client.ts#L364-L369

credential인자는 OAuthCredential 구현 객체로 받으며 직접 값을 줘서 객체를 만들거나 AuthApiClient 객체의 login 메서드를 이용해 로그인 한뒤 받아올 수 있습니다.
https://github.com/storycraft/node-kakao/blob/768cd3ae1659131ba8dfe8d3f5738afcf268bb62/examples/login-chat.ts#L43-L50

config 인자에 들어갈 수 있는 NodeKakao에서 사용하는 기본 config 객체는 이곳에서 볼 수 있습니다.
https://github.com/storycraft/node-kakao/blob/768cd3ae1659131ba8dfe8d3f5738afcf268bb62/src/config.ts#L7-L37

예제 코드 
```ts
const api = await AuthApiClient.create(DEVICE_NAME, DEVICE_UUID);
const loginRes = await api.login({
  email: EMAIL,
  password: PASSWORD,
});
if (!loginRes.success) throw new Error(`Web login failed with status: ${loginRes.status}`);

const client = await createSessionWebClient( 
   loginRes.result,
   DefaultConfiguration, 
   'https',
   'exampleapi.com',
 );
```

`SessionWebClient`객체는 카카오톡에서 사용하는 내부 api 호출 전용이며 다른 외부 api의 경우 동작하지 않을 수 있습니다.

**Reply by punch5545** [reply]
감사합니다. 질문 당시 선물하기 api를  호출했는데 401을 반환하는걸 보면 해당 api는 지원하지 않나봅니다.

![image](https://user-images.githubusercontent.com/28100137/133922857-a849e627-8b53-4215-9a52-d055b1505d0e.png)

**Reply by ggoek34** [reply]
혹시 해결되셨나요?

**Reply by punch5545** [reply]
선물하기 웹api는 동작을 안하네요.

**Reply by ggoek34** [reply]
답변감사드립니다. 
저도 찾고 있는데 안나오네요 ㅠㅠ
